sklearn/tests/test_common.py

```diff
@@ -63,8 +64,12 @@ def test_non_meta_estimators():
             continue
         if name.startswith("_"):
             continue
-        for check in _yield_all_checks(name, Estimator):
-            yield _named_check(check, name), name, Estimator
+        estimator = Estimator()
+        # check this on class
+        yield check_no_fit_attributes_set_in_init, name, Estimator
```

Comment:
If we just reframe `check_no_fit_attributes_set_in_init` as `check_no_fit_attributes_set_upon_clone`, perhaps we can get away with running it on an instance too.